src/lava/lib/optimization/solvers/bayesian/solver.py

```diff
@@ -0,0 +1,293 @@
+            limited to single objective
+        """
+        # validate input argument specifying the acquisition function
+        # config
+        valid_acq_funcs: list[str] = [
+            "LCB", "EI", "PI", "gp_hedge", "EIps", "PIps"
+        ]
+        self.acquisition_function_config: dict = Schema({
+            "type": And(
+                lambda x: x in valid_acq_funcs,
+                error=f"acq_func not in valid options: {valid_acq_funcs}"
+            )
+        }).validate(acq_func_config)
+
+        # validate input argument specifying the acquisition optimizer config
+        valid_acq_opts: list[str] = ["sampling", "lbfgs", "auto"]
+        self.acquisition_optimizer_config: dict = Schema({
+            "type": And(
+                lambda x: x in valid_acq_opts,
+                error=f"acq_opt not in valid options: {valid_acq_opts}"
+            )
+        }).validate(acq_opt_config)
+
+        # validate input argument specifying whether or not to plot results
+        self.enable_plotting: bool = Schema(
+            lambda x: type(x) == bool,
+            error="enable_plotting should be a boolean"
+        ).validate(enable_plotting)
+
+        # validate the type of specified surrogate regressor
+        valid_estimators: list[str] = ["GP"]
+        self.est_config: dict = Schema({
+            "type": And(
+                lambda x: x in valid_estimators,
+                error=f"estimator is not in valid options: {valid_estimators}"
+            ),
+        }).validate(est_config)
+
+        # validate the argued initial point generator
+        valid_ip_gens: list[str] = [
+            "random", "sobol", "halton", "hammersly", "lhs", "grid"
+        ]
+        self.ip_gen_config: str = Schema({
+            "type": And(
+                lambda x: x in valid_ip_gens,
+                error=f"ip generator is not in valid options: {valid_ip_gens}"
+            )
+        }).validate(ip_gen_config)
+
+        # validate the argued log directory
+        self.log_dir: str = Schema(
+            lambda x: os.path.isdir(x),
+            error=f"log_dir [{log_dir}] is not a directory"
+        ).validate(log_dir)
+
+        # validate the argued number of initial points
+        self.num_initial_points: int = Schema(
+            lambda x: type(x) == int and x > 0,
+            error="the number of initial points should be an int > 0"
+        ).validate(num_ips)
+
+        # validate the argued number of objectives
+        self.num_objectives: int = Schema(
+            lambda x: type(x) == int and x == 1,
+            error="num_objectives should be an int = 1"
+        ).validate(num_objectives)
+
+        # validate the argued seed
+        self.seed: int = Schema(
+            lambda x: type(x) == int,
+            error="random_state should be an integer"
+        ).validate(seed)
+
+    def solve(self, name: str, num_iter: int, problem: AbstractProcess,
```

Comment:
Same as the ctor. The actual function of this function is obscured by too much validation.